src/main/scala/strawman/collection/immutable/Set.scala

```diff
@@ -22,4 +23,15 @@ trait SetMonoTransforms[A, +Repr]
 
   def - (elem: A): Repr
 
+}
+
+trait SetPolyTransforms[A, +C[X] <: Set[X] with SetLike[X, C]]
+  extends collection.SetPolyTransforms[A, C] {
+
+  protected def coll: C[A]
+
+  def ++ (that: collection.Set[A]): C[A] =
+    if (that.isEmpty) coll
+    else that.foldLeft[C[A]](coll)(_ + _)
```

Comment:
This is the generic implementation of `++`. In contrast to the one inherited from `IterablePolyTransforms`, this one does not rely on the `View` + `fromIterable` combination.

I think I should check whether the `View` + `fromIterable` combination offers the same performances as this implementation or not.

A major drawback of this implementation is that it requires complex bounds for the type parameter `C[_]`.